cmd/vic-machine/converter/converter.go

```diff
@@ -0,0 +1,505 @@
+		return result, nil
+	}
+	err := convert(reflect.ValueOf(data), "", "", result)
+	return result, err
+}
+
+func convert(src reflect.Value, prefix string, tags reflect.StructTag, dest map[string][]string) error {
+	t := src.Type().String()
+	if converter, ok := typeConverters[t]; ok {
+		return converter(src, prefix, tags, dest)
+
+	}
+	if converter, ok := kindConverters[src.Kind()]; ok {
+		return converter(src, prefix, tags, dest)
+	}
+	log.Debugf("Skipping unsupported field, interface: %#v, kind %s", src, src.Kind())
+	return nil
+}
+
+func convertPtr(src reflect.Value, prefix string, tags reflect.StructTag, dest map[string][]string) error {
+	if src.IsNil() {
+		// no need to attempt anything
+		return nil
+	}
+	return convert(src.Elem(), prefix, tags, dest)
+}
+
+func convertStruct(src reflect.Value, prefix string, tags reflect.StructTag, dest map[string][]string) error {
+	defer trace.End(trace.Begin(fmt.Sprintf("prefix: %s, src: %s", prefix, src.String())))
+
+	// iterate through every field in the struct
+	for i := 0; i < src.NumField(); i++ {
+		field := src.Field(i)
+		// get field key, and keep going even the attribut key is empty, to make sure children attribute is not missing
+		tags := src.Type().Field(i).Tag
+		key := calculateKey(tags, prefix)
+		if err := convert(field, key, tags, dest); err != nil {
+			return err
+		}
+
+		if field.Kind() == reflect.Map {
+			// label handler is invoked in map converter
+			continue
+		}
+	}
+	return nil
+}
+
+func convertSlice(src reflect.Value, prefix string, tags reflect.StructTag, dest map[string][]string) error {
+	defer trace.End(trace.Begin(fmt.Sprintf("prefix: %s, src: %s", prefix, src)))
+
+	length := src.Len()
+	if length == 0 {
+		log.Debug("Skipping empty slice")
+		return nil
+	}
+
+	for i := 0; i < length; i++ {
+		if err := convert(src.Index(i), prefix, tags, dest); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func convertMap(src reflect.Value, prefix string, tags reflect.StructTag, dest map[string][]string) error {
+	defer trace.End(trace.Begin(fmt.Sprintf("prefix: %s, src: %s", prefix, src)))
+
+	// iterate over keys and recurse
+	mkeys := src.MapKeys()
+	length := len(mkeys)
+	if length == 0 {
+		log.Debug("Skipping empty map")
+		return nil
+	}
+
+	handler, hasHandler := labelHandlers[tags.Get(labelTag)]
+	// use tempMap to avoid duplicate processing
+	tempMap := make(map[string][]string)
+	for _, pkey := range src.MapKeys() {
+		if pkey.Kind() != reflect.String {
+			log.Errorf("Unsupported map key type interface: %s, kind %s", src, src.Kind())
+			continue
+		}
+		if !hasHandler {
+			if err := convert(src.MapIndex(pkey), prefix, tags, dest); err != nil {
+				return err
+			}
+			continue
+		}
+		if err := convert(src.MapIndex(pkey), prefix, tags, tempMap); err != nil {
+			return err
+		}
+		if err := handler(tempMap, pkey.String()); err != nil {
+			return err
+		}
+		for k, v := range tempMap {
+			addValues(dest, k, v)
+			delete(tempMap, k)
```

Comment:
hmm... for this one, it is. But I do have another case, need to replace the key in the map at https://github.com/vmware/vic/pull/5218/files/dfbf0e9799247d373e0b39d8c9b12d0bff667d24#diff-f1c87f067c3afedcd7836400c915a2b5R467